common/djangoapps/student/models.py

```diff
@@ -1894,29 +1894,64 @@ def refundable(self, user_already_has_certs_for=None):
     def refund_cutoff_date(self):
         """ Calculate and return the refund window end date. """
         # NOTE: This is here to avoid circular references
-        from openedx.core.djangoapps.commerce.utils import ecommerce_api_client, ECOMMERCE_DATE_FORMAT
-
+        from openedx.core.djangoapps.commerce.utils import ECOMMERCE_DATE_FORMAT
         date_placed = self.get_order_attribute_value('date_placed')
 
         if not date_placed:
             order_number = self.get_order_attribute_value('order_number')
             if not order_number:
                 return None
 
+            date_placed = self.get_order_attribute_from_ecommerce('date_placed')
+            if not date_placed:
+                return None
+
+            # also save the attribute so that we don't need to call ecommerce again.
+            username = self.user.username
+            enrollment_attributes = get_enrollment_attributes(username, str(self.course_id))
+            enrollment_attributes.append(
+                {
+                    "namespace": "order",
+                    "name": "date_placed",
+                    "value": date_placed,
+                }
+            )
+            set_enrollment_attributes(username, str(self.course_id), enrollment_attributes)
+
+        refund_window_start_date = max(
+            datetime.strptime(date_placed, ECOMMERCE_DATE_FORMAT),
+            self.course_overview.start.replace(tzinfo=None)
+        )
```

Comment:
Please add comments regarding what exactly we are doing here? Why we are replacing `tzinfo` with `None`?

Reply:
That's because we cannot compare a `tz-aware` object with a `tz-naive` object. See this for more details: https://stackoverflow.com/questions/15307623/cant-compare-naive-and-aware-datetime-now-challenge-datetime-end
BTW, this piece of code was not written by me, I'm just moving it to a different location.